chia/data_layer/data_store.py

```diff
@@ -990,18 +991,19 @@ async def delete(
         status: Status = Status.PENDING,
         root: Optional[Root] = None,
     ) -> Optional[Root]:
+        assert isinstance(key, bytes)
```

Comment:
We don't generally runtime type check and if we want to I'd consider a general library for that purpose.  It's certainly useful for double checking but I would rather not leave this in.